### PR TITLE
Upgrade library

### PR DESCRIPTION
**Pull request by areksvk**

### CODE REVIEW

File: src/Diagram.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+	}
+
+	_getRootProperties(){
+		return {
+			'algorithm': 'layered',
+			'direction':'RIGHT'
+		};
+    }
+
+    _setGraphSize(nodes) {
```
Comment (ondrejvelisek) [inline src/Diagram.js]: Wrong indentation here?

File: src/Component.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+	get className() {
+		return this._className;
+	}
+
+	get container() {
+		return this._container;
+	}
+
+    get observable() {
```
Comment (ondrejvelisek) [inline src/Component.js]: same indentation here?

Comment (mattesCZ) [reply src/Component.js]: Yes, we're mixing tabs and spaces. Please select one option and use it everywhere. ESLint should throw an error if configured...

File: src/DiagramEdges.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                .attr("class", "link")
+                .attr("stroke", EDGES_STROKE_COLOR)
+                .attr("stroke-width", 1)
+                .attr("fill", "transparent")
+                .attr("d", () => {
+                    const d = layout.edges[index].sections[0];
+                    let path = "";
+                    if (d.startPoint && d.endPoint) {
+                        path += "M" + d.startPoint.x + " " + d.startPoint.y + " ";
+                        (d.bendPoints || []).forEach(function (bp) {
```
Comment (mattesCZ) [inline src/DiagramEdges.js]: We're mixing _function_ with _arrow_ notation. Probably some copy-pasting? Not a big deal, but it's better to be consistent.

File: src/DiagramNodes.js [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+        return data.nodes.reduce((obj, item) => {
+            const edges = this._findEdgesRecursive(data.edges, [item.name]);
+            obj[item.name] = edges.map(edge => edge.end);
+            return obj;
+        }, {});
+    }
+
+    _findEdgesRecursive(edges, names, alreadySearched=[]) {
+        let results = [];
+        names.forEach(name => {
```
Comment (mattesCZ) [inline src/DiagramNodes.js]: I'd rather see it with `names.reduce()`, but it's just personal preference :)

File: src/DiagramNode.js [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+
+    setSelected(value) {
+        this._selected = value;
+    }
+
+    setSelectedSubsequent(value) {
+        this._selectedSubsequent = value;
+    }
+
+    setHighlighted(value) {
```
Comment (mattesCZ) [inline src/DiagramNode.js]: It's interesting, that these "public" methods like `setHighlighted` doesn't make any visual difference, until they're used together with `setStyle`. They just edit some internal property, am I right? Wouldn't it be better to "repaint" the node if I call some "public" method?

File: src/utils/Observable.js [hunk 6]
```diff
@@ -64,11 +76,11 @@ export default class Observable {
 
 	/**
 	 * @public
-	 * Destorys this observable, removes events and so on 
+	 * Destorys this observable, removes events and so on
```
Comment (mattesCZ) [inline src/utils/Observable.js]: Great, that there is no trailing whitespace anymore. Would be nice to fix the "destorys" typo as well :)

File: src/DiagramNode.js (hunk 5 above)
Comment (mattesCZ) [reply src/DiagramNode.js]: Maybe I'm missing some basic context. If it's used "publicly" only inside `DiagramNodes` component and it's not meant to be used directly externally, then it's probably fine to leave it like it is.

Comment (areksvk) [reply src/DiagramNode.js]: Exactly. it is public for purposes inside `DiagramNodes`.

**Comment by areksvk** [comment]
> While testing I've realized default colors looks half transparent. Is it Ok?

good point. Will be fixed with white baselayer.

**Comment by ondrejvelisek** [comment]
Not sure if there is some approve button but it looks good for me and can be merged. :-)

**Comment by mattesCZ** [comment]
I'm pretty sure I've seen such option, but I can't find it now... You can merge it @areksvk

**Pull request merged**